### PR TITLE
[gh-actions] Make yamllint strict and other changes

### PR DESCRIPTION
**Pull request by jstourac**
* Let's make yamllint strict
* Get rid of the yamllint GH action that may be out of date eventually
* Code linter warning fix into `sec-scan.yml`
* Update actions/checkout GH action to latest version

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by atheo89** [comment]
Thanks Jan, the changes /lgtm

**Review by rkpattnaik780 — APPROVED** [review]
/lgtm

File: .github/workflows/code-quality.yaml [hunk 1]
```diff
@@ -9,25 +9,26 @@ jobs:
   code-static-analysis:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4.1.1
```
Comment (jiridanek) [inline .github/workflows/code-quality.yaml:12]: I'd keep it @v4 and let it auto update

**Review by jiridanek — APPROVED** [review]

File: .github/workflows/code-quality.yaml (hunk 1 above)
Comment (jstourac) [reply .github/workflows/code-quality.yaml:12]: Right, wasn't aware - https://github.com/orgs/community/discussions/48058 . Thanks, I'll update.

**Review by jiridanek — APPROVED** [review]

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve 

:+1:

**Pull request merged**